lib/gui/models/store.js

```diff
@@ -287,10 +287,22 @@ const storeReducer = (state = DEFAULT_STATE, action) => {
         throw errors.createError('Missing image size');
       }
 
-      if (!_.isNumber(action.data.size)) {
+      if (!_.isPlainObject(action.data.size)) {
         throw errors.createError(`Invalid image size: ${action.data.size}`);
       }
 
+      if (!_.isNumber(action.data.size.original)) {
```

Comment:
Done! BTW, I'm making a note to myself to use something like JSON schema to decoratively validate these objects, so we don't have to write dozens of conditionals.